### PR TITLE
Sprint6 97 pojo domain

### LINKED ISSUE #97: As a MyLibrary developer, I want Catalog Service to publish POJOs for library domain objects.

### PR DESCRIPTION
**Pull request by kaladay**
resolves #97 

- Uses `edu.tamu.catalog` instead of `edu.tamu`.
- There are now three parts: `catalog-parent`, `catalog-domain`, and `catalog-service`.
- Utilizes Lombok.
- Renames **CatalogHolding** to **HoldingsRecord**.
- Use **Catalog Service** as the banner instead of **Weaver**.
- Tweak spring boot configuration to "skip", allowing for running `mvn clean spring-boot:run` from the parent directory.

### CODE REVIEW

**Comment by jsavell** [comment]
Will this be a problem for devops? My assumption has always been that they use this property, but I don't actually know.

https://github.com/TAMULib/CatalogService/pull/98/files#diff-237bb265cf2b00d7ce4da81d45d5038a4ae2763081e324a089fee6ab930cfc2eL26

**Comment by jsavell** [comment]
Just a note, if CatalogHolding gets renamed, we won't be able to put off refactoring GIFMService.

At minimum, we'll need to reference the updated name for the json key:
https://github.com/TAMULib/GIFMButtonService/blob/main/src/main/java/edu/tamu/app/service/CatalogService.java#L89

The Lombok conversion ups the urgency on converting GIFMService fully to the domain approach.

**Comment by wwelling** [comment]
> Just a note, if CatalogHolding gets renamed, we won't be able to put off refactoring GIFMService.
> 
> At minimum, we'll need to reference the updated name for the json key:
> https://github.com/TAMULib/GIFMButtonService/blob/main/src/main/java/edu/tamu/app/service/CatalogService.java#L89
> 
> The Lombok conversion ups the urgency on converting GIFMService fully to the domain approach.

I think it would be best to use Lombok and use domain approach as much as possible. Do you think the GIFMService refactor is within this sprint? Or should we defer refactoring CatalogHolding?

I would vote for best approach possible now for CatalogService. We either will have technical debt for GIFMService now or technical debt for both later.

**Comment by wwelling** [comment]
> Will this be a problem for devops? My assumption has always been that they use this property, but I don't actually know.
> 
> https://github.com/TAMULib/CatalogService/pull/98/files#diff-237bb265cf2b00d7ce4da81d45d5038a4ae2763081e324a089fee6ab930cfc2eL26

There will most likely require deployment updates for CatalogService. As the service itself is now a submodule, the war file will be in a different location. Will now be something like `service/target/catalog.war`. I don't see any reason the application.properties or other config files would require changing as they are templates placed in the correct place during deployment.

File: domain/src/main/java/edu/tamu/catalog/domain/model/HoldingsRecord.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package edu.tamu.catalog.domain.model;
+
+import java.util.Map;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@AllArgsConstructor
```
Comment (wwelling) [inline domain/src/main/java/edu/tamu/catalog/domain/model/HoldingsRecord.java:9]: For non entities, we can use [@Data lombok](https://projectlombok.org/features/Data) annotation. Is there any reason we do not want to declare the properties final?

File: pom.xml [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+  <groupId>edu.tamu.catalog</groupId>
+  <artifactId>catalog-parent</artifactId>
   <version>2.2.0</version>
-  <name>Catalog-Service</name>
-  <description>A service for interfacing with Library Catalogs</description>
+
+  <name>Catalog Parent</name>
+  <description>A service and a domain for interfacing with Library Catalogs</description>
 
   <parent>
```
Comment (wwelling) [inline pom.xml:11]: I would like to bring up to the team whether we should use weaver parent. CatalogService will not use anything from it and we should be getting away from the bad practices it prescribes.

**Review by wwelling — COMMENTED** [review]
Need to discuss refactoring the current CatalogService model and whether to use Weaver as a parent project or not.

**Comment by jsavell** [comment]
> 
> 
> > Just a note, if CatalogHolding gets renamed, we won't be able to put off refactoring GIFMService.
> > At minimum, we'll need to reference the updated name for the json key:
> > https://github.com/TAMULib/GIFMButtonService/blob/main/src/main/java/edu/tamu/app/service/CatalogService.java#L89
> > The Lombok conversion ups the urgency on converting GIFMService fully to the domain approach.
> 
> I think it would be best to use Lombok and use domain approach as much as possible. Do you think the GIFMService refactor is within this sprint? Or should we defer refactoring CatalogHolding?
> 
> I would vote for best approach possible now for CatalogService. We either will have technical debt for GIFMService now or technical debt for both later.

I just wanted to make sure we all understood that this was a breaking change for GIFMService and that we prepared for that.

I'm not opposed to the changes themselves and don't think this PR should wait on GIFMService to be updated.

**Comment by wwelling** [comment]
@jsavell sounds good. Let's move forward with the refactor and create GIFMService issue.

**Comment by kaladay** [comment]
> something



> > Will this be a problem for devops? My assumption has always been that they use this property, but I don't actually know.
> > https://github.com/TAMULib/CatalogService/pull/98/files#diff-237bb265cf2b00d7ce4da81d45d5038a4ae2763081e324a089fee6ab930cfc2eL26
> 
> There will most likely require deployment updates for CatalogService. As the service itself is now a submodule, the war file will be in a different location. Will now be something like `service/target/catalog.war`. I don't see any reason the application.properties or other config files would require changing as they are templates placed in the correct place during deployment.

At this time it uses `./service/target/ROOT.war` but only because the configuration has used `ROOT`.
I left that part alone, but I would be more than willing to rename `ROOT` to `CatalogService` (or `catalogservice`) such that the resulting war file is `./service/target/CatalogService.war`.
I believe this would simply be changing the `<finalName>ROOT</finalName>` from the `service/pom.xml` to `<finalName>CatalogService</finalName>`.

**Comment by wwelling** [comment]
@kaladay DevOps at one point requested all war files be ROOT.war so they could simplify the deployment and rename the file as they copied into Tomcat webapps.

**Comment by kaladay** [comment]
> @kaladay DevOps at one point requested all war files be ROOT.war so they could simplify the deployment and rename the file as they copied into Tomcat webapps.

This is why I tend to try to avoid making arbitrary changes.
I'm glad I left it alone then.

File: pom.xml (hunk 2 above)
Comment (jcreel) [reply pom.xml:11]: If we can drop it without consequence here, I would say we do so.

**Review by wwelling — APPROVED** [review]

**Pull request merged**